File: match-loader.rb
require 'json'
require 'byebug'
# Return an array of matches
class MatchLoader

  def self.get_matches
    file = File.read('matches.json')
    JSON.parse(file)['matches']
  end

  def self.get_team_gold_earned(participants)
    total_gold = 0
    participants.each do |participant|
      total_gold += participant['stats']['goldEarned']
    end
    total_gold
  end

  def self.get_team_kda(participants)
    kills, deaths, assists = 0, 0, 0
    participants.each do |participant|
      kills += participant['stats']['kills']
      deaths += participant['stats']['deaths']
      assists += participant['stats']['assists']
    end
    (kills + assists).to_f/deaths
  end

  def self.get_team_cs_rate(participants)
    cs_per_min = 0
    participants.each do |participant|
      cs_deltas = participant['timeline']['creepsPerMinDeltas']
      cs_per_min += (cs_deltas['zeroToTen'] + cs_deltas['tenToTwenty'])/2
    end
    cs_per_min
  end

  # Data format
  # Each data point is a hash as {x => [x1, x2, x3], y => 0 or 1}
  # Store all data points in an array
  def self.parse_data
    data = []
    matches = get_matches
    matches.each do |match|
      t1_dp, t2_dp = Hash.new, Hash.new
      t1_dp[:x], t2_dp[:x] = [], []

      team1 = match['participants'].take(5)
      t1_dp[:x] << get_team_gold_earned(team1)/match['matchDuration']
      t1_dp[:x] << get_team_kda(team1)
      t1_dp[:x] << get_team_cs_rate(team1)

      team2 = match['participants'].drop(5)
      t2_dp[:x] << get_team_gold_earned(team2)/match['matchDuration']
      t2_dp[:x] << get_team_kda(team2)
      t2_dp[:x] << get_team_cs_rate(team2)

      if match['teams'].first['winner']
        # team 1 is the winner
        t1_dp[:y] = 1
        t2_dp[:y] = 0
      else
        # team 2 is the winner
        t1_dp[:y] = 0
        t2_dp[:y] = 1
      end

      data << t1_dp
      data << t2_dp
    end
    data
  end
end

File: match_loader.py
import json

def get_matches():
    json_data = open('matches.json')
    matches = json.load(json_data)['matches']
    return matches # which is an array of matches

def get_team_gold_rate(team, match_duration):
    total_gold = 0
    for teammate in team:
        total_gold += teammate['stats']['goldEarned']
    return float(total_gold)/match_duration

def get_team_kda(team):
    kills, deaths, assists = 0, 0, 0
    for teammate in team:
        kills += teammate['stats']['kills']
        deaths += teammate['stats']['deaths']
        assists += teammate['stats']['assists']
    return float(kills + assists)/deaths

def get_team_cs_rate(team):
    cs_per_min = 0
    for teammate in team:
        cs_deltas = teammate['timeline']['creepsPerMinDeltas']
        cs_per_min += float(cs_deltas['zeroToTen'] + cs_deltas['tenToTwenty'])/2
    return cs_per_min

def get_parsed_data():
    parsed_data = []
    matches = get_matches()
    for match in matches:
        team_1_data, team_2_data = {}, {}
        team_1_data['x'], team_2_data['x'] = [], []

        team_1 = match['participants'][0:5]
        team_1_data['x'].append(1)
        team_1_data['x'].append(get_team_gold_rate(team_1, match['matchDuration']))
        team_1_data['x'].append(get_team_kda(team_1))
        team_1_data['x'].append(get_team_cs_rate(team_1))

        team_2 = match['participants'][5:10]
        team_2_data['x'].append(1)
        team_2_data['x'].append(get_team_gold_rate(team_2, match['matchDuration']))
        team_2_data['x'].append(get_team_kda(team_2))
        team_2_data['x'].append(get_team_cs_rate(team_2))

        if match['teams'][0]['winner']:
            team_1_data['y'] = 1
            team_2_data['y'] = 0
        else:
            team_1_data['y'] = 0
            team_2_data['y'] = 1
        parsed_data.append(team_1_data)
        parsed_data.append(team_2_data)
    return parsed_data
